File: TA/Frontend/RHStoreWS/RHStorePrendasBO/PromocionBO.cs
using RHStoreBaseBO;
using RHStoreBaseBO.ServiciosWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStorePrendasBO
{
	public class PromocionBO : BaseBO
	{
        public int insertar(string nombre, string descripcion, double valorDescuento, tipoPromocion tipo, DateTime fechaInicio, DateTime fechaFin, prenda[] prendasSeleccionadas, trabajador _trabajador)
        {
            return PromocionWS.insertarPromocion(nombre, descripcion, valorDescuento, tipo, fechaInicio, fechaFin, prendasSeleccionadas, _trabajador);
        }
        public int modificar(int idPromocion, string nombre, string descripcion, double valorDescuento, tipoPromocion tipo, DateTime fechaInicio, DateTime fechaFin, prenda[] prendasSeleccionadas, trabajador _trabajador)
        {
            return PromocionWS.modificarPromocion(idPromocion, nombre, descripcion, valorDescuento, tipo, fechaInicio, fechaFin, prendasSeleccionadas, _trabajador);
        }
        public int eliminar(int idPromocion)
        {
            return PromocionWS.eliminarPromocion(idPromocion);
        }

        public BindingList<promocion> listarTodos()
        {
            return new BindingList<promocion>(PromocionWS.listarPromocionesTodas());
        }

        public promocion obtenerPorId(int idPromocion)
        {
            return PromocionWS.obtenerPromocionPorId(idPromocion);
        }

		public BindingList<promocion> listarPorNombreDescripcion(string cadena)
		{
			return new BindingList<promocion>(PromocionWS.listarPromocionesPorNombreDescripcion(cadena));
		}

		public BindingList<prenda> listarPrendaAgregadas(int idPromocion)
		{
			// Obtener la lista de prendas desde el servicio y convertir a List<prenda>
			List<prenda> prendas = PromocionWS.listarPrendasAgregadas(idPromocion)?.ToList();

			// Si prendas es null, inicializar con una lista vacía
			return new BindingList<prenda>(prendas ?? new List<prenda>());
		}
	}
}

File: TA/Backend/SoftRhRMI_Servidor/src/pe/edu/pucp/softrh/rmi/interfaces/PromocionBO.java
package pe.edu.pucp.softrh.rmi.interfaces;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.prendas.model.Prenda;
import pe.edu.pucp.softrh.prendas.model.Promocion;
import pe.edu.pucp.softrh.prendas.model.TipoPromocion;
import pe.edu.pucp.softrh.usuarios.model.Trabajador;

public interface PromocionBO extends Remote{
    Integer insertar(String nombre, String descripcion, Double valorDescuento, TipoPromocion tipo, Date fechaInicio, Date fechaFin, ArrayList<Prenda> prendasSeleccionadas, Trabajador trabajador) throws RemoteException;

    Integer modificar(Integer idPromocion, String nombre, String descripcion, Double valorDescuento, TipoPromocion tipo, Date fechaInicio, Date fechaFin, ArrayList<Prenda> prendasSeleccionadas, Trabajador trabajador) throws RemoteException;

    Integer eliminar(Integer idPromocion) throws RemoteException;

    ArrayList<Promocion> listarTodos() throws RemoteException;

    Promocion obtenerPorId(Integer idPromocion) throws RemoteException;

    ArrayList<Promocion> listarPorNombreDescripcion(String cadena) throws RemoteException;

    ArrayList<Prenda> listarPrendasAgregadas(int idPromocion) throws RemoteException;
}
